pkg/cover/backend/elf.go

```diff
@@ -417,23 +424,65 @@ func readCoverPoints(file *elf.File, info *symbolInfo) ([2][]uint64, error) {
 	if err != nil {
 		return pcs, fmt.Errorf("failed to read .text: %v", err)
 	}
-	end := len(data) - callLen + 1
+	end := len(data)
+	if arch == targets.AMD64 {
+		end = end - callLen + 1
+	}
 	for i := 0; i < end; i++ {
-		pos := bytes.IndexByte(data[i:end], 0xe8)
+		pos := bytes.IndexByte(data[i:end], opcode)
 		if pos == -1 {
 			break
 		}
 		pos += i
 		i = pos
-		off := uint64(int64(int32(binary.LittleEndian.Uint32(data[pos+1:]))))
 		pc := text.Addr + uint64(pos)
-		target := pc + off + callLen
+		var off uint64
+		var target uint64
+		if arch == targets.AMD64 {
+			off = uint64(int64(int32(binary.LittleEndian.Uint32(data[pos+1:]))))
+			target = pc + off + uint64(callLen)
+		} else if arch == targets.ARM64 {
+			b := data[pos-callLen+1 : pos]
+			b = append(b, 0x00)
+			off = uint64(int64(int32(binary.LittleEndian.Uint32(b))))
+			off = 4 * off
+			target = pc + off - uint64(callLen) + 1
+			pc -= 3
+		}
 		if target == info.tracePC {
 			pcs[0] = append(pcs[0], pc)
 		} else if info.traceCmp[target] {
 			pcs[1] = append(pcs[1], pc)
 		}
 	}
+	if arch == targets.ARM64 {
+		opcode = 0x97
+		for i := 0; i < end; i++ {
+			pos := bytes.IndexByte(data[i:end], opcode)
+			if pos == -1 {
+				break
+			}
+			pos += i
+			i = pos
+			pc := text.Addr + uint64(pos)
+			b := data[pos-callLen+1 : pos]
+			b = append(b, 0xff)
+			off := uint64(math.Abs(float64(int64(int32(binary.LittleEndian.Uint32(b))))))
```

Comment:
Use of floating point in binary data processing makes me very nervous. I also can't understand what exactly it's doing (requires knowledge of floating point representation). What does it do on binary level? Can this be replaced with some bit and/or/xor?